### DISCUSSION TITLE
umi max中layout中配置菜单模式为顶部导航失效

### DISCUSSION THREAD

**Discussion by lyj2209914083** [umijs/umi#11760] [Q&A]
在一个完全新建的Umi Max项目中，在设置中设置layout为‘top’顶部导航，但是界面仍然是侧边导航。
设置示意图：
<img width="1063" alt="image" src="https://github.com/umijs/umi/assets/70557762/25de3bf0-ad82-4f39-a849-34906564856a">
界面效果示意图：
<img width="1364" alt="image" src="https://github.com/umijs/umi/assets/70557762/c480a36b-cefa-4f51-8cff-6c050934a71c">

**Comment by fz6m** [comment]
这里是 umi 配置文件，属于静态配置，在 `src/app.ts` 还有运行时配置，优先级高于这个，会覆盖这部分静态配置，自己检查下吧，建议都配置到动态运行时配置里，因为这里的静态配置写法有限。

**Reply by lyj2209914083** [reply]
我在app.ts中并没有配置相关属性，按理来说配置文件中的顶层导航配置应该生效
<img width="681" alt="image" src="https://github.com/umijs/umi/assets/70557762/f31bda6c-21eb-45d6-8f16-6ba779344529">

**Reply by fz6m** [reply]
如有问题，需要给一个最小复现。

**Reply by lyj2209914083** [reply]
最小复现仓库地址： https://github.com/lyj2209914083/umi_max_demo

**Reply by fz6m** [reply]
目前静态 [`layout.xxx`](https://umijs.org/docs/max/layout-menu#%E9%85%8D%E7%BD%AE) 配置只支持以下属性：`title` / `formatMessage` / `locale` ，其他 layout 组件的配置均需通过 `src/app.ts` 中 `export const layout` 运行时进行配置，配置项可参考 [ProLayout](https://procomponents.ant.design/components/layout) 。

**Comment by sp4rkw** [comment]
直接layout: 'top',
<img width="1160" alt="image" src="https://github.com/umijs/umi/assets/33112152/e16bc2d7-88a5-4cff-a979-0081bbb4cacc">
注意格式的区别

我也被这个坑了半天，可能文档都觉得大家非常容易理解吧
